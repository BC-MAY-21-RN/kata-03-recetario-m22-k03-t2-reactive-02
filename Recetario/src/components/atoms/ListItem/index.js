import React from 'react';
import {Text, Image, StyleSheet, TouchableOpacity} from 'react-native';

const ListItem = props => {
  const url = props.url;
  console.log(url);
  return (
    <TouchableOpacity>
      <Image
        style={styles.image}
        source={{
          uri: 'https://cdn3.iconfinder.com/data/icons/vol-5/128/shirt-128.png',
        }}
      />
      <Text style={styles.text}>{props.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  image: {
    width: 100,
    height: 100,
  },
  text: {
    fontSize: 20,
    fontWeight: 'bold',
  },
});

export default ListItem;
