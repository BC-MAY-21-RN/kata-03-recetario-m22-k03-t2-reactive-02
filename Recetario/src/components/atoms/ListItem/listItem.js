import React from 'react';
import {Text, Image, StyleSheet, TouchableOpacity} from 'react-native';

const ListItem = props => {
  const {pantalla, secondStyle, titulo} = props;
  return (
    <TouchableOpacity
      style={secondStyle ? styles.container2 : styles.container}
      onPress={() => {
        pantalla.navigate('Detalle', {
          id: props.id,
          titulo: titulo,
        });
      }}>
      <Image
        style={secondStyle ? styles.image2 : styles.image}
        source={{uri: props.url}}
      />
      <Text style={styles.text}>{props.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  image: {
    width: 130,
    height: 130,
    borderRadius: 10,
  },
  image2: {
    width: 145,
    height: 175,
    borderRadius: 10,
  },
  text: {
    fontSize: 15,
    fontWeight: 'bold',
    color: 'white',
    textAlign: 'center',
    paddingTop: 5,
  },
  container: {
    margin: 5,
    height: 200,
    width: 130,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 15,
  },
  container2: {
    margin: 5,
    height: 220,
    width: 145,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 15,
  },
});

export default ListItem;
