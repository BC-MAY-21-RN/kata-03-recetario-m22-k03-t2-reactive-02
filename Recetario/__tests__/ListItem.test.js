import 'react-native';
import React from 'react';
import ListItem from '../src/components/atoms/ListItem/listItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<ListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
