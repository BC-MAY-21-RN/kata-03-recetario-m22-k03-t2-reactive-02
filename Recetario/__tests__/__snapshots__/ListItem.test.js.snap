// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 200,
      "margin": 5,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 15,
      "opacity": 1,
      "width": 130,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 130,
        "width": 130,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 15,
        "fontWeight": "bold",
        "paddingTop": 5,
        "textAlign": "center",
      }
    }
  />
</View>
`;
